lib/setup.py

```diff
@@ -4,10 +4,17 @@
 import subprocess
 import sys
 
-from pipenv.project import Project
-from pipenv.utils import convert_deps_to_pip
 from setuptools.command.install import install
 
+try:
+    from pipenv.project import Project
+    from pipenv.utils import convert_deps_to_pip
+except:
+    exit_msg = (
+        "pipenv is required to package Streamlit. Please install pipenv and try again"
+    )
+    sys.exit(exit_msg)
```

Comment:
Would love to make it so that we can exit the build with a nicer message display but I think all the ugliness comes from `pip install ...`. Tried to have it print and pass but then `pip` complains about a missing files instead. Would love suggestions/workarounds that can improve our error display.

Reply:
Haha this looks fine to me~